core/src/test/scala/org/apache/spark/sql/delta/stats/StatsCollectionSuite.scala

```diff
@@ -127,9 +127,9 @@ class StatsCollectionSuite
         val statsDf = statsDF(deltaLog)
         assert(statsDf.where('numRecords.isNotNull).count() > 0)
         checkAnswer(statsDf, Seq(
-          Row(4, Row(10, 20), Row(17, 27)),
-          Row(2, Row(11, 21), Row(19, 29)),
-          Row(4, Row(12, 22), Row(18, 28))))
+          Row(3, Row(10, 20), Row(19, 29)),
+          Row(4, Row(12, 22), Row(17, 27)),
+          Row(3, Row(11, 21), Row(18, 28))))
```

Comment:
@zsxwing Is it a regression for Delta?

Reply:
@wangyum do you know if there is any change in Spark `repartition`, such as changing the hash partitioning logic? Looks like the `repartition` behavior is slightly difference. Previously, we got 3 files and the numbers of records are  (4, 2, 4). But now we got a different distribution (3, 4, 3).

Reply:
cc @cloud-fan

Reply:
IIUC, the change here is a row may get a different hash value and write to a different file in 3.3.1?

Reply:
Looks like it's related to this commit: https://github.com/apache/spark/commit/2bae604d05ef734ed4562be5fc545d5e4abfbbd2

Reply:
hm, this seems okay. I cannot think of any correctness issues for this change. Thanks for finding the commit.

Reply:
I forgot that `repartition` is using `RoundRobinPartitioning`.